core/src/toga/command.py

```diff
@@ -228,6 +243,36 @@ def __init__(
         self._enabled = True
         self.enabled = enabled
 
+    @classmethod
+    def standard(cls, app, id, **kwargs):
+        """Create an instance of a standard command for the provided app.
+
+        :param app: The application instance for which the command is being created.
+        :param id: The ID of the standard command to create.
+        :param kwargs: Overrides for any default properties of the standard command.
+            Accepts the same arguments as the :class:`~toga.Command` constructor.
+        """
+        # The value of the ID constant is the method on the app instance
+        cmd_kwargs = {"id": id}
+        try:
+            cmd_kwargs["action"] = simple_handler(getattr(app, id))
```

Comment:
Making `Command.standard` give its commands a default action is a good idea, but I think we can make some improvements:

* All the default actions should correspond to a public method, or a simple combination of them, so the developer can call them in other ways.
* The documentation of each command should state its default action, with links to the corresponding methods.

Also, as we agreed that ["the size of the `toga.App` surface is a concern"](https://github.com/beeware/toga/pull/2666#issuecomment-2229759096), can we move the document-related methods into the DocumentSet class? That means the standard commands' methods would be as follows:

* ABOUT – `App.about`
* EXIT – `App.on_exit`, then `App.exit` if it returns true
* NEW – `DocumentSet.new`
* OPEN – `DocumentSet.open` – calling it with no arguments could display an Open dialog like the current `App._open`.
* PREFERENCES – no default action, and the documentation should say so.
* SAVE – `App.current_window.save`
* SAVE_ALL – `DocumentSet.save_all`
* SAVE_AS – `App.current_window.save_as`
* VISIT_HOMEPAGE – `App.visit_homepage`

Finally, since this would move some code into DocumentSet, this would be a good opportunity to move the DocumentSet and DocumentWindow classes into documents.py, because app.py and window.py are now around 1000 lines each.

Reply:
I agree that DocumentSet and DocumentWindow can definitely move into documents.py (and, for that matter, WindowSet into Window). I've made that change.

I agree that it would be desirable to move logic like the save handlers into a location other than the base App class; the complication is how to *find* those methods. As currently defined, the value of the standard command constants is just the method that needs to be `getattr()`ed - but that only works because they're all attributes of the app instance. Moving that functionality onto other classes means at the very least we need to include the class as part of the descriptor.... but I'll need to see if there are any other consequences/limitations.

I also generally agree that the methods being invoked should be public; the two major complications are `open` and `exit`.

The issue with `open` is mostly naming - we need a naming convention for "request a user action to occur" that is distinction from the `open()` method that is the actual file opening method.

`exit` is currently implemented with `_request_exit`, because there's a need to dereference the potentially async on_exit handler. Maybe the answer in both cases is to add a `request_X` method (making the existing method public, and renaming `_open` to `request_open`.

Reply:
Moving the document methods turned out to be easier than I thought. `standard()` requires that you pass in a context object - so we can pass in the app's documentSet as the context object, and move the document creation/save methods to be instance methods on DocumentSet.

Reply:
...or better still: pass in app, but provide a fully qualified lookup path to the method as the constant.

Reply:
Looks good, just a couple more comments:

* The documentation of each standard command should have a link to the method called by its default action, or for PREFERENCES, a statement that there is no default action.
* Could we move `App.document_types` to `DocumentSet.types`, so all the document stuff is contained in that one object?

Reply:
I've added links to the default actions, and a note about PREFERENCES.

I'm not sure moving `document_types` to `DocumentSet` makes as much sense, though. I get the inclination to put all the document management stuff in one namespace; but to me, `app.documents.types` reads more as the types of the specific document instances that have been opened, whereas `app.document_types` reads more as intended - the types of document this app can manage.

Reply:
I'm not sure why I posted the `document_types` comment here; I'll move it to a separate thread below so this one can be resolved once you push the commands docstring updates.